sagemathinc/cocalc
# Decaffeinate misc, opts, and immutable-types. Add Jest testing

# Description
Convert these coffee files to JS to enable more testing.

The coffeescript tests remain unchanged.

Also added Jest.

# Testing Steps
1. Make sure $NODE_ENV = development
1. run `node_modules/.bin/mocha test/types.coffee`
1. run `node_modules/.bin/mocha test/misc-test.coffee`
1. run `jest` to see the few Jest tests run
1. Smoke test, did some interacting with projects: making files, deleting files, using a course.

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
